### ISSUE TITLE
Sometimes do  not hear correct directional alert?

### ISSUE THREAD

**Issue by terracoda** [phetsims/friction#149]
@zepumph, I don't know if yo have noticed this, but sometimes I hear left or right when I press up or down.

At first, I just thought it might be a delay thing, but it is not. I very carefully press and I can get the incorrect direction.

I think it happens most often for vertical movement. For example, I just pressed down and I heard "right". And since the model thought I went right, I did not hear the additional rubbing cue, "Rub fast or slow."

Not sure what could be causing this?
I will try to add some test cases.

**Comment by zepumph** [comment]
@terracoda I cannot reproduce this if only using the up and down arrows. I can though do this if in the same press I hold both a vertical and horizontal arrow key while the book goes from one corner to its opposite corner.

The reason this is happening is because the direction algorithm is compiling an angle based on the start and end points. Given the picture below, if the start location of the book is the center, I marked out what range the end point has to be to get each directions. Sometimes you get both, but most often it compiles them into single directions. Note that this is the algorithm that was used in BASE. Because friction is much more of a horizontal sim, I am not surprised that more often than not the complex movements (with multiple keys) come out as just a "left" or "right" alert. You can see that "cardinal" (primary) directions get twice as much space on the circle as diagonal ones.

Regions are sectioned out by strong pen marks intersecting the book at a perpendicular angle.
![20181123_114720](https://user-images.githubusercontent.com/6856943/48959970-91e06c80-ef26-11e8-9d3a-32a084312249.jpg)

@terracoda does this make sense? I'm not sure what is the best way forward. Ideally we would have the same definition of a direction for all sims.

**Comment by emily-phet** [comment]
Hmm - maybe the double regions (e.g., left & up, right & up) need to be wider?

**Comment by terracoda** [comment]
I did further testing last night and this morning and I can still get incorrect direction alerts. It seems to happen on the Zoomed-in book. I have noticed it there.

**Comment by terracoda** [comment]
@zepumph, the circle graph makes sense and very familiar from BASE. 
I do not know why I get incorrect directions sometimes.

**Comment by zepumph** [comment]
assigning to myself to try to reproduce.

**Comment by terracoda** [comment]
@zepumph, I've obviously only experienced this issue in Safari with Voice Over.

**Comment by zepumph** [comment]
>@zepumph, I've obviously only experienced this issue in Safari with Voice Over.

Can you reproduce it without VO turned on, just by looking at the a11y view?

**Comment by terracoda** [comment]
I'll try.

**Comment by terracoda** [comment]
I only hit the Left Arrow key and I got 2 alerts "up", "left".

<img width="574" alt="evidence" src="https://user-images.githubusercontent.com/1902116/49111618-f1c17500-f290-11e8-82d6-96caff168ebb.png">

**Comment by terracoda** [comment]
After a few more tries I got another random weird one.
I pressed the Left Arrow and got a "right" alert. 
<img width="357" alt="more_evidence" src="https://user-images.githubusercontent.com/1902116/49111937-c68b5580-f291-11e8-872a-f3a1f81dd698.png">

It can happen on the actual book, too.
Here I pressed Left Arrow, but got "right".
<img width="363" alt="happens-on-book-too" src="https://user-images.githubusercontent.com/1902116/49112443-11f23380-f293-11e8-8470-67e08d68d0a5.png">

**Comment by zepumph** [comment]
@terracoda and I were able to discuss this, and saw that it is always after a reset all button press. Here is one walkthrough I was able to reproduce:

press left
alert "left"
reset
press left 
alert "right"
reset
press left 
alert "move down to rub harder"
reset

**Comment by zepumph** [comment]
In the above commit there were 4 bug fixes, but only of two of them were manifesting in Friction.

Bugs fixed for friction:
1. The "last location" of the book was not reset either. This location is compared to the new location (through the unit circle posted above) to figure out what direction to alert next. This is the bug that @terracoda was noticing for a long time. 
2. LeftRightAlertPair (the ones that keep track of left/right movement alerts to know when to alert "MoveDownToRubHarder") was not reset with `resetAllButton`.


other bug fixes:
3/4: Both `MovementDescriber.movementAlerts` and `BorderAlertsDescriber`'s alerts accepted `Utterance` as a potential arg, but those Utterance instances weren't being reset on reset.

@terracoda please review.

**Comment by terracoda** [comment]
This looks good @zepumph. I am no longer able to reproduce the incorrect directional alerts after a reset.

But I see to have some inconsistencies after a break away. Is it possible you need to clear the last location after a break away as well? I seem to get incorrect directional alerts when I move down after a break away. 

After a break away, I often hear two directions with one key press down, or an incorrect direction, e.g. "left" instead of "down".

**Comment by terracoda** [comment]
@zepumph, reassigning back to you. 

To reproduce,
1. rub until you get a break away
2. carefully press down arrow or S key
3. you should hear the last horizontal key press that was close to the break away (left or right) rather than "down"

Let me know if you cannot reproduce.

**Comment by zepumph** [comment]
This is fixed now. @terracoda please review

**Comment by terracoda** [comment]
Verified! Thanks @zepumph, closing this issue.
